src/Microsoft.VisualStudio.Editors/MyApplication/MyApplicationProperties.vb

```diff
@@ -1388,15 +1388,20 @@ Namespace Microsoft.VisualStudio.Editors.MyApplication
         ''' <remarks></remarks>
         Friend Shared Function IsMySubMainSupported(Hierarchy As IVsHierarchy) As Boolean
             Try
-                Dim obj As Object = Nothing
-                If Hierarchy IsNot Nothing Then
-                    VSErrorHandler.ThrowOnFailure(Hierarchy.GetProperty(VSITEMID.ROOT, __VSHPROPID.VSHPROPID_BrowseObject, obj))
+                ' Query the MyType and OutputType properties from the project properties
+                Dim project As Project = EnvDTEProject(Hierarchy)
+
+                If project Is Nothing Then
+                    Return False
                 End If
-                Dim props3 As VSLangProj80.VBProjectProperties3 = TryCast(obj, VSLangProj80.VBProjectProperties3)
-                If props3 IsNot Nothing Then
+
+                Dim myTypeProperty As [Property] = GetProjectProperty(project, NameOf(VSLangProj80.VBProjectProperties3.MyType))
```

Comment:
This method is pretty inefficient, it walks every property, when it can just use the string indexer of the Properties instance.

Reply:
Logged #5060